community/rfcs/24-08-02-OPEA-AIAvatarChatbot.md

```diff
@@ -0,0 +1,140 @@
+<p align="left">
+  <img src="assets/avatar4.png" alt="Image 1" width="100"/>
+  <img src="assets/avatar1.jpg" alt="Image 2" width="100"/>
+  <img src="assets/avatar2.jpg" alt="Image 3" width="100"/>
+  <img src="assets/avatar3.png" alt="Image 4" width="100"/>
+  <img src="assets/avatar5.png" alt="Image 5" width="100"/>
+  <img src="assets/avatar6.png" alt="Image 6" width="100"/>
+</p> 
+
+The chatbot will:
+* Be able to understand and respond to user text and audio queries, with a backend LLM model
+* Synchronize audio response chunks with image/video frames, to generate a high-quality video of the avatar speaking in real-time
+* Present the animated avatar response to the user in a user-friendly UI
+* Use multimodal retrieval-augmented generation (RAG) to generate more accurate, in-domain responses, in v0.2
+
+New microservices include:
+* animation 
+
+New megaservices include:
+* AvatarChatbot
+
+## Motivation
+<!-- List why this problem is valuable to solve? Whether some related work exists? -->
+* Enterprises in medical, finance, education, entertainment, etc. industries are increasingly adopting AI chatbots to improve customer service and user experience. Yet existing OPEA applications (ChatQnA, AudioQnA, SearchQnA, etc.) are text-based and lack interactive visual elements.
+* Enterprises look for multimodal AI solutions that can process both audio and visual inputs, to build lip-synchronized and face-animated chatbots that are more engaging and human-like.
+* This RFC aims to fill these gaps by introducing a new microservice, animation, that can be integrated seamlessly with existing micro- and mega-services in OPEA, to enhance the platform's capabilities in multimodal AI, human-computer interaction, and digital human graphics.
+
+Overall, this project adds to the OPEA platform a new microservice block that animates the chatbot appearance, and integrates it with the existing chatbot pipelines such as [ChatQnA](https://github.com/opea-project/GenAIExamples/tree/2e312f44edbcbf89bf00bc21d9e9c847405ecae8/ChatQnA), [AudioQnA](https://github.com/opea-project/GenAIExamples/tree/2e312f44edbcbf89bf00bc21d9e9c847405ecae8/AudioQnA), [SearchQnA](https://github.com/opea-project/GenAIExamples/tree/2e312f44edbcbf89bf00bc21d9e9c847405ecae8/SearchQnA), etc., to build new chatbot megaservices that can interact with users in a more human-like way.
+
+Related works include [Nvidia Audio2Face](https://docs.nvidia.com/ace/latest/modules/a2f-docs/index.html), [Lenovo Deepbrain AI Avatar](https://www.deepbrain.io/ai-avatars), [BitHuman](https://www.bithuman.io/), etc.
+
+## Design Proposal
+<!-- This is the heart of the document, used to elaborate the design philosophy and detail proposal. -->
+
+### Avatar Chatbot design
+[Proposal slides](https://intel.sharepoint.com/:p:/s/mlconsultingandsupport/EecmTa5Ca61Pm7ES6vlZtykBtgbJL_fuhipfu_yybIlBig?e=HDS0VF)
+
+<img src="assets/design.png" alt="Avatar Chatbot design" width="800"/>
```

Comment:
the Design doesn't match Flowchat.
Could we align both?  Moveover, what different messages between those two diagrams?

Reply:
The difference is that the Design includes RAG feature, that will be released in v0.2.  
The Flowchart is based on v0.1.
Explained accordingly in the revision.